Library/Homebrew/cask/audit.rb

```diff
@@ -770,32 +770,48 @@ def audit_reverse_migration
     end
 
     sig { void }
-    def audit_https_availability
-      return unless download
-
-      if cask.url && !cask.url.using
-        validate_url_for_https_availability(cask.url, "binary URL", cask.token, cask.tap,
-                                            location: cask.url.location,
-                                            user_agents: [cask.url.user_agent], referer: cask.url&.referer)
-      end
-
-      if cask.livecheckable? && !cask.livecheck.url.is_a?(Symbol)
-        validate_url_for_https_availability(cask.livecheck.url, "livecheck URL", cask.token, cask.tap,
-                                            check_content: true)
-      end
-
-      return unless cask.homepage
+    def audit_homepage_https_availability
+      return unless online?
+      return unless (homepage = cask.homepage)
 
       user_agents = if cask.tap&.audit_exception(:simple_user_agent_for_homepage, cask.token)
         ["curl"]
       else
         [:browser, :default]
       end
 
-      validate_url_for_https_availability(cask.homepage, SharedAudits::URL_TYPE_HOMEPAGE, cask.token, cask.tap,
-                                          user_agents:   user_agents,
-                                          check_content: true,
-                                          strict:        strict?)
+      validate_url_for_https_availability(
+        homepage, SharedAudits::URL_TYPE_HOMEPAGE,
+        user_agents:   user_agents,
+        check_content: true,
+        strict:        strict?
```

Comment:
Is there actually a reason to only make the `homepage` audit strict instead of all of them?

Reply:
Ah, it's passed to `curl_check_http_content` for additional checks and not actually used for audit severity.